tests/unit_tests/test_parsers.py

```
@@ -145,3 +145,67 @@ def test_read_pepxml(real_pepxml):
+# Multi-file and error-handling tests ---------------------------------------------
+def test_read_tide_two_separate_files(target_tide_txt, decoy_tide_txt):
+    """Passing two files concatenates their PSMs into one dataset."""
+    psms = crema.read_tide([target_tide_txt, decoy_tide_txt])
+    assert psms.data.shape[0] == 20  # 10 targets + 10 decoys
+    assert psms.targets.sum() == 10
+    assert (~psms.targets).sum() == 10
+
+
+def test_read_tide_target_only_raises(target_tide_txt):
+    """A file containing only target PSMs must raise ValueError (no decoys)."""
+    with pytest.raises(ValueError, match="[Nn]o decoy"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)